iothub/device/src/Transport/HttpTransportHandler.cs

```diff
@@ -427,6 +421,29 @@ private static TimeSpan GetInvokeDeviceMethodOperationTimeout(MethodInvokeReques
             return customHeaders;
         }
 
+        private static object ConvertToMessageSystemProperty(string messagePropertyName, IEnumerable<string> headerValue)
+        {
+            string propertyValue = headerValue?.First();
+
+            if (messagePropertyName.Equals(MessageSystemPropertyNames.EnqueuedTime, StringComparison.Ordinal)
```

Comment:
Why not use a switch statement ?

Reply:
no reason! I had started out with 2 properties, so `if` block was cleaner. I have 4 custom implementations in here now, so I can change it to switch case.